ext/mysql2/mysql2_ext.h

```diff
@@ -12,15 +12,21 @@ void Init_mysql2(void);
 #ifdef HAVE_MYSQL_H
 #include <mysql.h>
 #include <mysql_com.h>
+#include <mysql_version.h>
 #include <errmsg.h>
 #include <mysqld_error.h>
 #else
 #include <mysql/mysql.h>
 #include <mysql/mysql_com.h>
+#include <mysql/mysql_version.h>
 #include <mysql/errmsg.h>
 #include <mysql/mysqld_error.h>
 #endif
 
+#if 50708 <= MYSQL_VERSION_ID && MYSQL_VERSION_ID < 100000
+#define HAVE_MYSQL_TYPE_JSON
+#endif
```

Comment:
Rather than keying off of a version check, we can either check if the symbol is defined in extconf.rb, or even simpler, `#ifdef MYSQL_TYPE_JSON`